src/main/java/io/r2dbc/h2/codecs/ClobCodec.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.h2.codecs;
+
+import io.r2dbc.h2.util.Assert;
+import io.r2dbc.spi.Clob;
+import org.h2.value.Value;
+import org.h2.value.ValueLobDb;
+import reactor.core.publisher.Mono;
+
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+final class ClobCodec extends AbstractCodec<Clob> {
+
+    ClobCodec() {
+        super(Clob.class);
+    }
+
+    @Override
+    boolean doCanDecode(int dataType) {
+        return dataType == Value.CLOB;
+    }
+
+    @Override
+    Clob doDecode(Value value, Class<? extends Clob> type) {
+        ValueLobDb clobValue = (ValueLobDb) value.convertTo(Value.CLOB);
```

Comment:
Actually it may be not only the `ValueLobDb`, but also a legacy `ValueLob`, if database file was created by some really old version of H2, it's better to avoid cast to `ValueLobDb` here.